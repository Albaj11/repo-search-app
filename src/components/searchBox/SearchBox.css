.topSection {
  background-color: #f1f1f1;
  padding: 10px;
}

.content {
  background-color: #ddd;
  padding: 10px;
}

#topHeader {
  font-family: Georgia, serif;
  font-size: 46px;
  text-align: center;
}

p {
  font-family: Verdana, sans-serif;
  font-size: 16px;
  text-align: center;
}

.center {
  margin: auto;
  /* width: 80%;
  border: 3px solid green; */
  padding: 10px;
}

.SearchFieldCenter {
  margin: auto;
  width: 30%;
  padding: 10px;
}

#mainTable {
  font-family: Arial, Helvetica, sans-serif;
  border-collapse: collapse;
  width: 100%;
}

#mainTable td,
.mainTable th {
  border: 1px solid #ddd;
  padding: 8px;
}

#mainTable tr:nth-child(even) {
  background-color: #f2f2f2;
}

#mainTable tr:hover {
  background-color: #ddd;
}

#mainTable th {
  padding-top: 12px;
  padding-bottom: 12px;
  text-align: center;
  background-color: #2196f3;
  color: white;
  border: 1px solid grey;
}

.mainForm {
  margin: auto;
  max-width: 300px;
}

/* Style the search field */
form.mainForm input[type="text"] {
  padding: 10px;
  font-size: 17px;
  border: 1px solid grey;
  float: left;
  background: #f1f1f1;
}

/* Style the submit button */
form.mainForm button {
  float: left;
  width: 20%;
  padding: 10px;
  background: #2196f3;
  color: white;
  font-size: 17px;
  border: 1px solid grey;
  border-left: none; /* Prevent double borders */
  cursor: pointer;
}

form.mainForm button:hover {
  background: #0b7dda;
}

/* Clear floats */
form.mainForm::after {
  content: "";
  clear: both;
  display: table;
}

.container {
  position: relative;
  width: 100%;
}

.image {
  display: block;
  width: 100%;
  height: auto;
}

.overlay {
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: #008cba;
  overflow: hidden;
  width: 100%;
  height: 0;
  transition: 0.5s ease;
}

.container:hover .overlay {
  height: 100%;
}

.text {
  white-space: nowrap;
  color: white;
  font-size: 10px;
  position: absolute;
  overflow: hidden;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
}
