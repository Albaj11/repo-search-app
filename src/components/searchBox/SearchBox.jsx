import React, { Component } from "react";
import Table from "../table/Table";
import "./SearchBox.css";

let searchTerm;
export default class SearchBox extends Component {
  constructor(props) {
    super(props);

    this.state = {
      repositories: [],
      isUsername: false,
    };
  }

  onClick = (e) => {
    searchTerm = this.searchBox.value;
    // let epoint = `https://api.github.com/users${searchTerm}/repos`;
    let endpoint =
      "https://api.github.com/search/repositories?sort=stars&order=desc&q=" +
      searchTerm;
    fetch(endpoint)
      .then((blob) => blob.json())
      .then((response) => {
        this.setState({ repositories: response.items, isUsername: true });
      });
    e.preventDefault();
  };

  render() {
    console.log(this.props);
    return (
      <React.Fragment>
        <div className="topSection" data-testid="searchBox-1">
          <h1 id="topHeader">Search your repository</h1>
          <p>
            A repository contains all the files of your project, including the
            revision history.
          </p>
        </div>

        <div className="content">
          <form className="mainForm">
            <input
              type="text"
              placeholder="Enter username"
              className="searchbox"
              ref={(input) => {
                this.searchBox = input;
              }}
            />
            <button onClick={this.onClick}>
              <i className="fa fa-search"></i>
            </button>
          </form>
        </div>
        <Table
          repositories={this.state.repositories}
          isUsername={this.state.isUsername}
        />
      </React.Fragment>
    );
  }
}
