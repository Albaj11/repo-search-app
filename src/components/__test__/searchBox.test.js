import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import SearchBox from "../searchBox/SearchBox";

afterEach(() => {
  cleanup();
});

test("should render searchBox component", () => {
  render(<SearchBox />);
  const searchBoxElement = screen.getAllByTestId("searchBox-1");
});

test("match snapshot", () => {
  const tree = renderer.create(<SearchBox />).toJSON();
  expect(tree).toMatchSnapshot();
});
