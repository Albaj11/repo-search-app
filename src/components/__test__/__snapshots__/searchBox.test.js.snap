// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
Array [
  <div
    className="topSection"
    data-testid="searchBox-1"
  >
    <h1
      id="topHeader"
    >
      Search your repository
    </h1>
    <p>
      A repository contains all the files of your project, including the revision history.
    </p>
  </div>,
  <div
    className="content"
  >
    <form
      className="example"
      style={
        Object {
          "margin": "auto",
          "maxWidth": "300px",
        }
      }
    >
      <input
        className="searchbox"
        placeholder="Enter username"
        type="text"
      />
      <button
        onClick={[Function]}
      >
        <i
          className="fa fa-search"
        />
      </button>
    </form>
  </div>,
]
`;

exports[`match snapshot 2`] = `
Array [
  <div
    className="topSection"
    data-testid="searchBox-1"
  >
    <h1
      id="topHeader"
    >
      Search your repository
    </h1>
    <p>
      A repository contains all the files of your project, including the revision history.
    </p>
  </div>,
  <div
    className="content"
  >
    <form
      className="example"
      style={
        Object {
          "margin": "auto",
          "maxWidth": "300px",
        }
      }
    >
      <input
        className="searchbox"
        placeholder="Enter username"
        type="text"
      />
      <button
        onClick={[Function]}
      >
        <i
          className="fa fa-search"
        />
      </button>
    </form>
  </div>,
]
`;

exports[`match snapshot table component 1`] = `
Array [
  <div
    className="topSection"
    data-testid="searchBox-1"
  >
    <h1
      id="topHeader"
    >
      Search your repository
    </h1>
    <p>
      A repository contains all the files of your project, including the revision history.
    </p>
  </div>,
  <div
    className="content"
  >
    <form
      className="example"
      style={
        Object {
          "margin": "auto",
          "maxWidth": "300px",
        }
      }
    >
      <input
        className="searchbox"
        placeholder="Enter username"
        type="text"
      />
      <button
        onClick={[Function]}
      >
        <i
          className="fa fa-search"
        />
      </button>
    </form>
  </div>,
]
`;
